Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m

```diff
@@ -291,14 +299,53 @@ - (void)adsManagerDidRequestContentPause:(nonnull IMAAdsManager *)adsManager {
 }
 
 - (void)adsManagerDidRequestContentResume:(nonnull IMAAdsManager *)adsManager {
-    // record content-resume first so adbreakend happens before playing (brackets the ad break
+    // record content-resume first so adbreakend happens before playing (brackets the ad break)
     [self onContentPauseOrResume:false];
     if (self.customerAdsManagerDelegate) {
         [self.customerAdsManagerDelegate adsManagerDidRequestContentResume:adsManager];
     }
 }
 
-/* IMAAdsLoaderDelegate */
+#pragma mark IMAAdsManagerDelegate optional methods
+
+- (void)adsManager:(IMAAdsManager *)adsManager
+adDidProgressToTime:(NSTimeInterval)mediaTime
+         totalTime:(NSTimeInterval)totalTime {
+    [self trySafely:^{
```

Comment:
Obj-C has a built in way to do the kind of check that `trySafely` does by [calling `respondsToSelector`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector) which returns a boolean.

Reply:
As in the PR text, we need the customer delegates to be instances NSObject. They don't, so there's no way to use respondsToSelector

The main question I have is whether we should accept things as they are and use this approach, or should we require customers' delegates to also inherit from NSObject

Reply:
Or even a secret third option that I'm not aware of

Reply:
Yeah it's typical to have to use `respondsToSelector:` with a delegate that doesn't explicitly inherit from `NSObject`. Cast the delegate to a bare `id` just for the check: `[(id)delegate respondsToSelector:@selector(...])`